### PR TITLE
[doc] Fix broken internal documention links

### PR DESCRIPTION
**Pull request by HU90m**
This fixes broken links in the documentation. Broken links were found using  [`lychee`](https://github.com/lycheeverse/lychee).

```sh
cargo install lychee
lychee . -o /tmp/output.md -f markdown --require-https --exclude-all-private --cache --exclude-path sw/vendor/ --exclude-path hw/vendor/ --exclude-path site/ --exclude-path hw/ip_templates
```

*Note: Currently links between books have to be URLs. Lychee can't check for this.*

These changes are split over two PRs to make reviews easier. This PR has the large number of fixes to internal links. The second PR, https://github.com/lowRISC/opentitan/pull/17655, fewer more miscellaneous changes.

Fixes: https://github.com/lowRISC/opentitan/issues/17588

### CODE REVIEW

File: hw/top_earlgrey/ip_autogen/alert_handler/doc/checklist.md [hunk 1]
```diff
@@ -225,11 +225,11 @@ Tests         | [SIM_SEC_CM_VERIFIED][]                 | Done        |
 Coverage      | [SIM_COVERAGE_REVIEWED][]               | Done        |
 Review        | [SEC_CM_DV_REVIEWED][]                  | Done        |
 
-[SEC_CM_TESTPLAN_COMPLETED]:          ../../../../doc/project_governance/checklist/README.md#sec_cm_testplan_completed
-[FPV_SEC_CM_VERIFIED]:                ../../../../doc/project_governance/checklist/README.md#fpv_sec_cm_verified
-[SIM_SEC_CM_VERIFIED]:                ../../../../doc/project_governance/checklist/README.md#sim_sec_cm_verified
-[SIM_COVERAGE_REVIEWED]:              ../../../../doc/project_governance/checklist/README.md#sim_coverage_reviewed
-[SEC_CM_DV_REVIEWED]:                 ../../../../doc/project_governance/checklist/README.md#sec_cm_dv_reviewed
+[SEC_CM_TESTPLAN_COMPLETED]:          ../../../../../doc/project_governance/checklist/README.md#sec_cm_testplan_completed
+[FPV_SEC_CM_VERIFIED]:                ../../../../../doc/project_governance/checklist/README.md#fpv_sec_cm_verified
```
Comment (jwnrt) [inline hw/top_earlgrey/ip_autogen/alert_handler/doc/checklist.md:229]:
```suggestion
[FPV_SEC_CM_PROVEN]:                ../../../../../doc/project_governance/checklist/README.md#fpv_sec_cm_proven
```

File: hw/top_earlgrey/ip_autogen/rv_plic/doc/checklist.md [hunk 2]
```diff
@@ -53,23 +53,23 @@ Code Quality  | [CDC_SYNCMACRO][]       | N/A         |
...
+[FEATURE_FROZEN]:      ../../../../../doc/project_governance/checklist/README.md#feature_frozen
+[FEATURE_COMPLETE]:    ../../../../../doc/project_governance/checklist/README.md#feature_complete
+[AREA_CHECK]:          ../../../../../doc/project_governance/checklist/README.md#area_check
+[PORT_FROZEN]:         ../../../../../doc/project_governance/checklist/README.md#port_frozen
+[ARCHITECTURE_FROZEN]: ../../../../../doc/project_governance/checklist/README.md#architecture_frozen
+[REVIEW_TODO]:         ../../../../../doc/project_governance/checklist/README.md#review_todo
+[STYLE_X]:             ../../../../../doc/project_governance/checklist/README.md#style_x
+[LINT_PASS]:           ../../../../../doc/project_governance/checklist/README.md#lint_pass
+[CDC_SETUP]:           ../../../../../doc/project_governance/checklist/README.md#cdc_setup
+[FPGA_TIMING]:         ../../../../../doc/project_governance/checklist/README.md#fpga_timing
```
Comment (jwnrt) [inline hw/top_earlgrey/ip_autogen/rv_plic/doc/checklist.md:69]:
```suggestion
[TIMING_CHECK]:         ../../../../../doc/project_governance/checklist/README.md#timing_check
```

File: hw/top_earlgrey/ip_autogen/rv_plic/doc/checklist.md [hunk 3]
```diff
@@ -53,23 +53,23 @@ Code Quality  | [CDC_SYNCMACRO][]       | N/A         |
...
+[CDC_SYNCMACRO]:       ../../../../../doc/project_governance/checklist/README.md#cdc_syncmacro
+[SEC_CM_DOCUMENTED]:   ../../../../../doc/project_governance/checklist/README.md#sec_cm_documented
+[SEC_RND_CNST]:        ../../../../../doc/project_governance/checklist/README.md#sec_rnd_cnst
```
Comment (jwnrt) [inline hw/top_earlgrey/ip_autogen/rv_plic/doc/checklist.md:72]: I don't see anything like this for `D2`, remove?
```suggestion
```

File: hw/top_earlgrey/ip_autogen/rv_plic/doc/checklist.md [hunk 4]
```diff
@@ -194,26 +194,26 @@ Issues        | [ALL_LOW_PRIORITY_ISSUES_ROOT_CAUSED][] | Done        |
...
+[SIM_FW_SIMULATED]:                   ../../../../../doc/project_governance/checklist/README.md#sim_fw_simulated
+[SIM_NIGHTLY_REGRESSION_V2]:          ../../../../../doc/project_governance/checklist/README.md#sim_nightly_regression_v2
+[SIM_CODE_COVERAGE_V2]:               ../../../../../doc/project_governance/checklist/README.md#sim_code_coverage_v2
+[SIM_FUNCTIONAL_COVERAGE_V2]:         ../../../../../doc/project_governance/checklist/README.md#sim_functional_coverage_v2
+[FPV_CODE_COVERAGE_V2]:               ../../../../../doc/project_governance/checklist/README.md#fpv_code_coverage_v2
+[FPV_COI_COVERAGE_V2]:                ../../../../../doc/project_governance/checklist/README.md#fpv_coi_coverage_v2
+[PRE_VERIFIED_SUB_MODULES_V2]:        ../../../../../doc/project_governance/checklist/README.md#pre_verified_sub_modules_v2
+[NO_HIGH_PRIORITY_ISSUES_PENDING]:    ../../../../../doc/project_governance/checklist/README.md#no_high_priority_issues_pending
+[ALL_LOW_PRIORITY_ISSUES_ROOT_CAUSED]:../../../../../doc/project_governance/checklist/README.md#all_low_priority_issues_root_caused
```
Comment (jwnrt) [inline hw/top_earlgrey/ip_autogen/rv_plic/doc/checklist.md:214]:
```suggestion
[ALL_LOW_PRIORITY_ISSUES_ROOT_CAUSED]: ../../../../../doc/project_governance/checklist/README.md#all_low_priority_issues_root_caused
```
Weirdly this missing space didn't break the link, it just meant `README.md` was translated to `README.html`, not `index.html`. Probably a bug in that preprocessor.

Don't forget to re-align the others if you add a space here

File: hw/top_earlgrey/ip_autogen/rv_plic/doc/checklist.md [hunk 5]
```diff
@@ -225,11 +225,11 @@ Tests         | [SIM_SEC_CM_VERIFIED][]                 | Not Started |
 Coverage      | [SIM_COVERAGE_REVIEWED][]               | Not Started |
 Review        | [SEC_CM_DV_REVIEWED][]                  | Not Started |
 
-[SEC_CM_TESTPLAN_COMPLETED]:          ../../../../doc/project_governance/checklist/README.md#sec_cm_testplan_completed
-[FPV_SEC_CM_VERIFIED]:                ../../../../doc/project_governance/checklist/README.md#fpv_sec_cm_verified
-[SIM_SEC_CM_VERIFIED]:                ../../../../doc/project_governance/checklist/README.md#sim_sec_cm_verified
-[SIM_COVERAGE_REVIEWED]:              ../../../../doc/project_governance/checklist/README.md#sim_coverage_reviewed
-[SEC_CM_DV_REVIEWED]:                 ../../../../doc/project_governance/checklist/README.md#sec_cm_dv_reviewed
+[SEC_CM_TESTPLAN_COMPLETED]:          ../../../../../doc/project_governance/checklist/README.md#sec_cm_testplan_completed
+[FPV_SEC_CM_VERIFIED]:                ../../../../../doc/project_governance/checklist/README.md#fpv_sec_cm_verified
```
Comment (jwnrt) [inline hw/top_earlgrey/ip_autogen/rv_plic/doc/checklist.md:229]:
```suggestion
[FPV_SEC_CM_PROVEN]:                ../../../../../doc/project_governance/checklist/README.md#fpv_sec_cm_proven
```

File: hw/top_earlgrey/ip_autogen/rv_plic/doc/checklist.md [hunk 6]
```diff
@@ -251,16 +251,16 @@ Issues        | [NO_ISSUES_PENDING][]             | Not Started |
...
-[X_PROP_ANALYSIS_COMPLETED]:     ../../../../doc/project_governance/checklist/README.md#x_prop_analysis_completed
-[FPV_ASSERTIONS_PROVEN_AT_V3]:   ../../../../doc/project_governance/checklist/README.md#fpv_assertions_proven_at_v3
-[SIM_NIGHTLY_REGRESSION_AT_V3]:  ../../../../doc/project_governance/checklist/README.md#sim_nightly_regression_at_v3
-[SIM_CODE_COVERAGE_AT_100]:      ../../../../doc/project_governance/checklist/README.md#sim_code_coverage_at_100
+[DESIGN_DELTAS_CAPTURED_V3]:     ../../../../../doc/project_governance/checklist/README.md#design_deltas_captured_v3
+[X_PROP_ANALYSIS_COMPLETED]:     ../../../../../doc/project_governance/checklist/README.md#x_prop_analysis_completed
+[FPV_ASSERTIONS_PROVEN_AT_V3]:   ../../../../../doc/project_governance/checklist/README.md#fpv_assertions_proven_at_v3
+[SIM_NIGHTLY_REGRESSION_AT_V3]:  ../../../../../doc/project_governance/checklist/README.md#sim_nightly_regression_at_v3
+[SIM_CODE_COVERAGE_AT_100]:      ../../../../../doc/project_governance/checklist/README.md#sim_code_coverage_at_100
 [SIM_FUNCTIONAL_COVERAGE_AT_100]:../../../../doc/project_governance/checklist/README.md#sim_functional_coverage_at_100
```
Comment (jwnrt) [inline hw/top_earlgrey/ip_autogen/rv_plic/doc/checklist.md:259]:
```suggestion
[SIM_FUNCTIONAL_COVERAGE_AT_100]: ../../../../doc/project_governance/checklist/README.md#sim_functional_coverage_at_100
```

As above, this missing space causes the `README` -> `index` preprocessor to fail.

**Review by jwnrt — CHANGES_REQUESTED** [review]
Great work with this, must have taken ages. Did you automate some of these, or do them manually?

I've clicked every link and added comments for any that didn't work right, but many of them are just `#anchors` that don't got to the right places. Some of these are more important than others, so up to you how many you update.

Thanks!

**Comment by hcallahan-lowrisc** [comment]
What is your opinion of 'lychee' having used it now? It would be great to have something like this in CI to check links when rebuilding the site. Is it worth pursing a lychee-based solution to do this?

File: hw/ip/usbdev/dv/README.md [hunk 7]
```diff
@@ -21,7 +21,9 @@ For detailed information on USBDEV design features, please see the [USBDEV HWIP
 USBDEV testbench has been constructed based on the [CIP testbench architecture](../../../dv/sv/cip_lib/README.md).
 
 ### Block diagram
-![Block diagram](tb.svg)
+<!--
+![Block diagram](./doc/tb.svg)
+-->
```
Comment (hcallahan-lowrisc) [inline hw/ip/usbdev/dv/README.md:26]: nit: delete

File: hw/ip_templates/alert_handler/README.md
Comment (hcallahan-lowrisc) [inline hw/ip_templates/alert_handler/README.md:1]: A general comment on the ip_templates/ip_autogen files based on what I learned in https://github.com/lowRISC/opentitan/pull/17467
Currently the `make -k -C hw top` command run as part of the lints re-generates the ip_autogen components, but the documentation step is simple and does not check links are correct or recalculate them based on the location of the generated files.
Hence, I think we need to ensure that relative links within the `ip_templates` markdown is correct for its rendered destination. For links within the templated hwip docs, this is easy, but links to other parts of the repo may not be correct with the same relative paths.

**Review by hcallahan-lowrisc — COMMENTED** [review]
Nice job! This is great to be getting these things fixed up. I agree with the comments by @jwnrt in his review.
Does the comment about the `ip_templates/ip_autogen` docs make sense? I think what needs to happen is to make changes to the `ip_templates` code, then run `make -k -C hw top` to propogate those changes to the `ip_autogen` files.

File: hw/ip_templates/alert_handler/README.md
Comment (hcallahan-lowrisc) [reply hw/ip_templates/alert_handler/README.md:1]: https://github.com/lowRISC/opentitan/pull/17467#issuecomment-1463792471

File: doc/guides/getting_started/src/SUMMARY.md [hunk 8]
```diff
@@ -8,7 +8,7 @@
 - [Formal Verification](setup_formal.md)
 - [Building (and Testing) Software](build_sw.md)
 - [Building Documentation](build_docs.md)
-- [Running e2e Tests](https://opentitan.org/book/sw/device/silicon_creator/rom/doc/e2e_tests.md)
+- [Running e2e Tests](https://opentitan.org/book/sw/device/silicon_creator/rom/doc/e2e_tests.html)
```
[the remark this answers is not in the document]
Comment (HU90m) [reply doc/guides/getting_started/src/SUMMARY.md:11]: Looks like I noticed it at that time https://github.com/lowRISC/opentitan/pull/17483#discussion_r1140195384

File: doc/security/specs/secure_boot/README.md [hunk 9]
```diff
@@ -93,7 +93,7 @@ If the key indicated in the manifest has a role that doesn't match the lifecycle
 All of these keys are 3072-bit RSA public keys with exponent e=65537 (the “F4 exponent”).
 
 The `ROM` has `N` key slots (the exact number depends on the `ROM`) numbered from `0` to `N-1`.
-The `CREATOR_SW_CFG_SIGVERIFY_RSA_KEY_EN` item in the [OTP](otp-mmap) can be used to invalidate a key at
```
[the remark this answers is not in the document]
Comment (HU90m) [reply doc/security/specs/secure_boot/README.md:96]: My best guess was that it was a placeholder until there was a nice OTP memory map page to point to.

File: hw/top_earlgrey/dv/README.md [hunk 10]
```diff
@@ -143,4 +143,4 @@ For a list of available tests  to run, please see the 'Tests' column in the [tes
 
 ## Testplan (Gate level simulations)
 Note that the descriptions of the test below may be replicated from the table above.
-[Testplan](../data/chip_testplan.hjson:gls)
+[Testplan](../data/chip_testplan.hjson)
```
[the remark this answers is not in the document]
Comment (HU90m) [reply hw/top_earlgrey/dv/README.md:146]: Ah no, I'll add this to this testplan to the summary.md

File: hw/ip_templates/alert_handler/README.md
Comment (HU90m) [reply hw/ip_templates/alert_handler/README.md:1]: Ah yes, thank you. I've edited the `ip_templates` and generated the rest.

File: hw/top_earlgrey/ip_autogen/rv_plic/doc/checklist.md (hunk 6 above)
Comment (HU90m) [reply hw/top_earlgrey/ip_autogen/rv_plic/doc/checklist.md]: done

File: hw/top_earlgrey/ip_autogen/alert_handler/dv/README.md [hunk 11]
```diff
@@ -12,34 +12,34 @@
...
-* [TileLink host interface](../../../dv/sv/tl_agent/README.md)
+* [Clock and reset interface](../../../../dv/sv/common_ifs/README.md)
+* [TileLink host interface](../../../../dv/sv/tl_agent/README.md)
 * ALERT_HANDLER IOs
-* Alerts and escalations([`alert_esc_if`](../../../dv/sv/alert_esc_agent/README.md))
-* Interrupts ([`pins_if`](../../../dv/sv/common_ifs/README.md))
-* Devmode ([`pins_if`](../../../dv/sv/common_ifs/README.md))
+* Alerts and escalations([`alert_esc_if`](../../../../dv/sv/alert_esc_agent/README.md))
+* Interrupts ([`pins_if`](../../../../dv/sv/common_ifs/README.md))
+* Devmode ([`pins_if`](../../../../dv/sv/common_ifs/README.md))
```
[the remark this answers is not in the document]
Comment (HU90m) [reply hw/top_earlgrey/ip_autogen/alert_handler/dv/README.md:35]: Yep :smiley:

**Comment by HU90m** [comment]
> What is your opinion of 'lychee' having used it now? It would be great to have something like this in CI to check links when rebuilding the site. Is it worth pursing a lychee-based solution to do this?

Yeah, I like it. It doesn't check anchors, which would be nice to have for links between markdown files, but this could be contributed upstream in https://github.com/lycheeverse/lychee/issues/185 .

We could have a check only for local links in CI and a full checks (including external links) in the nightlies.

File: hw/top_earlgrey/ip_autogen/rv_plic/doc/checklist.md (hunk 3 above)
Comment (HU90m) [reply hw/top_earlgrey/ip_autogen/rv_plic/doc/checklist.md]: Looks like it was moved to D2S (and renamed) in commit `f28ef61435fd41b546acc1f200371eda4f555032`.

**Review by jwnrt — APPROVED** [review]
Very nice work! LGTM

**Comment by hcallahan-lowrisc** [comment]
> > What is your opinion of 'lychee' having used it now? It would be great to have something like this in CI to check links when rebuilding the site. Is it worth pursing a lychee-based solution to do this?
> 
> Yeah, I like it. It doesn't check anchors, which would be nice to have for links between markdown files, but this could be contributed upstream in [lycheeverse/lychee#185](https://github.com/lycheeverse/lychee/issues/185) .
> 
> We could have a check only for local links in CI and a full checks (including external links) in the nightlies.

In that case we should create a follow-up issue to consider adding CI jobs using lychee. I think it would be really valuable.

**Comment by hcallahan-lowrisc** [comment]
One other comment, when you rebase can you seperate it from any new changes? I try to rebase the old branch, comment 'rebased', then force-push the new changes. It's basically impossible to review otherwise with the GitHub UI. Thanks!

**Review by hcallahan-lowrisc — APPROVED** [review]
Otherwise, LGTM!

**Comment by HU90m** [comment]
> One other comment, when you rebase can you seperate it from any new changes? I try to rebase the old branch, comment 'rebased', then force-push the new changes. It's basically impossible to review otherwise with the GitHub UI. Thanks!

Sorry, I don't quite follow. I'll come and ask you offline

**Review by andreaskurth — APPROVED** [review]
Very important and thorough fixes, thanks @HU90m.

LGTM, even though I did not check every link, from glancing over all changes and the systematic testing approach you took, this looks mature

+1 on adding a CI check that catches broken links

**Review by engdoreis — APPROVED** [review]

**Pull request merged**